ludwig/hyperopt/execution.py

```diff
@@ -27,11 +29,33 @@
 
 try:
     import ray
+    from ray.util.queue import Queue as RayQueue
     from ray import tune
-    from ray.tune.utils import wait_for_gpu
     from ray.tune import register_trainable
+    from ray.tune.suggest import BasicVariantGenerator, ConcurrencyLimiter
+    from ray.tune.syncer import get_cloud_sync_client
+    from ray.tune.utils import wait_for_gpu
+    from ray.tune.utils.placement_groups import PlacementGroupFactory
+    from ludwig.backend.ray import RayBackend, RayRemoteTrainer
 except ImportError:
     ray = None
+    get_horovod_kwargs = None
+
+    class RayBackend:
+        pass
+
+    class RayRemoteTrainer:
+        pass
+
+
+def _is_ray_backend(backend) -> bool:
```

Comment:
In general, it would be nice if we could avoid calling `_is_ray_backend()` in so many places. This could be a big refactor, so feel free to just add a TODO. But basically my preference would be to encapsulate this behavior behind an interface, like a `HyperoptBackend` or similar.

Reply:
Added a TODO note